### PR TITLE
New landing page

### PR DESCRIPTION
**Pull request by melissawm**
# Description

New landing page including a "Latest News" section pulled directly from README.md. 

The page should work in both light and dark mode (inspired by docs.jaxstack.ai).

The news section will be pulled in form whatever is in between the `<!-- NEWS START -->` and `<!-- NEWS END -->` comments in the README file, so these need to be kept there in the future.

# Tests

Documentation builds correctly and a preview can be seen here: https://melissawm.github.io/maxtext/

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,86 +18,92 @@
...
-* \[July 16, 2025\] We will be restructuring the MaxText repository for improved organization and clarity. Please review the [proposed structure](https://github.com/AI-Hypercomputer/maxtext/blob/main/RESTRUCTURE.md) and provide feedback.  
-* \[July 11, 2025\] Multi-Token Prediction (MTP) training support\! Adds an auxiliary loss based on predicting multiple future tokens, inspired by [DeepSeek-V3 paper](https://arxiv.org/html/2412.19437v1), to enhance training efficiency.  
-* \[June 25, 2025\] DeepSeek R1-0528 variant is now supported.  
+* \[September 5, 2025\] MaxText has moved to an `src` layout as part of [RESTRUCTURE.md](https://github.com/AI-Hypercomputer/maxtext/blob/main/RESTRUCTURE.md). For existing environments, please run `pip install -e .` from MaxText root.
```
Comment (bvandermoon) [inline README.md]: Nit: This wasn't introduced in this PR, but we should include a specific commit in this URL and the equivalent one below. We will delete this file after the work is completed so the link will break at that point

File: docs/index.html [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        <ul>
+            <li><a href="http://jax.dev/">JAX</a> - core array operations and program transformations</li>
+            <li><a href="https://flax.readthedocs.io/en/latest/">Flax</a> - For building neural networks</li>
+            <li><a href="https://orbax.readthedocs.io/en/latest/">Orbax</a> - For checkpointing and persistence utilities</li>
+            <li><a href="https://optax.readthedocs.io/en/latest/">Optax</a> - For gradient processing and optimization</li>
+            <li><a href="https://tunix.readthedocs.io/en/latest/">Tunix</a> - A JAX-native LLM Post-Training Library</li>
+            <li><a href="https://github.com/jax-ml/ml_dtypes">ml_dtypes</a> - NumPy dtype extensions for machine learning.</li>
+            <li>Optional data loading libraries (<a href="https://google-grain.readthedocs.io/en/latest/">Grain</a> or <a href="https://www.tensorflow.org/guide/data">tf.data</a>)</li>
+        </ul>
+    </section>
```
Comment (bvandermoon) [inline docs/index.html:43]: Is it just me or does this banner look too wide on the page you shared? I am ok with it if you think it is the right styling, just wanted to call it out

Comment (melissawm) [reply docs/index.html:43]: Yeah I had that impression yesterday too. I will fix

File: docs/index.html [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            <h3>High-performance</h3>
+            <p>MaxText achieves high Model FLOPs Utilization (MFU) and tokens/second from single host to very large clusters while staying simple and largely "optimization-free" thanks to the power of JAX and the XLA compiler.</p>
+        </div>
+        <div>
+            <h3>Pre-training</h3>
+            <p>MaxText provides opinionated implementations for how to achieve optimal performance across a wide variety of dimensions like sharding, quantization, and checkpointing.</p>
+        </div>
+        <div>
+            <h3>Post-training</h3>
+            <p>MaxText provides a scalable framework using Tunix</p>
```
Comment (jacoguzo) [inline docs/index.html]: Replace Post Training description with: "MaxText provides a scalable framework to fine-tune proprietary or OSS models using state-of-the-art Reinforcement Learning (RL) algorithms (e.g., GRPO) and techniques (e.g. SFT, Knowledge Distillation, etc)."

File: docs/index.html [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    </section>
+    <section class="banner">
+        <h3>JAX AI Stack</h3>
+        <p>The JAX AI Stack is a curated collection of libraries that researchers and engineers, both inside and outside of Google, have found useful for implementing and deploying the models behind generative AI tools like Imagen, Gemini, and more.</p>
+        <ul>
+            <li><a href="http://jax.dev/">JAX</a> - core array operations and program transformations</li>
+            <li><a href="https://flax.readthedocs.io/en/latest/">Flax</a> - For building neural networks</li>
+            <li><a href="https://orbax.readthedocs.io/en/latest/">Orbax</a> - For checkpointing and persistence utilities</li>
+            <li><a href="https://optax.readthedocs.io/en/latest/">Optax</a> - For gradient processing and optimization</li>
+            <li><a href="https://tunix.readthedocs.io/en/latest/">Tunix</a> - A JAX-native LLM Post-Training Library</li>
```
Comment (jacoguzo) [inline docs/index.html]: Replace Tunix description w/ : 
 <li><a 
href="https://tunix.readthedocs.io/en/latest/">Tunix</a> - A JAX Library with the latest experimental algorithms and post-training techniques </li>

File: docs/index.html [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            <li><a href="https://github.com/jax-ml/ml_dtypes">ml_dtypes</a> - NumPy dtype extensions for machine learning.</li>
+            <li>Optional data loading libraries (<a href="https://google-grain.readthedocs.io/en/latest/">Grain</a> or <a href="https://www.tensorflow.org/guide/data">tf.data</a>)</li>
+        </ul>
```
Comment (jacoguzo) [inline docs/index.html:42]: Could we add these sections of the stack as well? 

[MaxText model library](https://maxtext.readthedocs.io/en/latest/index.html#model-library) for JAX LLMs highly optimized for TPUs
[vLLM on TPU ](https://blog.vllm.ai/2025/10/16/vllm-tpu.html)for high performance sampling (inference) for Reinforcement Learning (RL)
[Pathways](https://docs.cloud.google.com/ai-hypercomputer/docs/workloads/pathways-on-cloud/pathways-intro) for multi-host inference (sampling) and highly efficient weight transfer

**Comment by melissawm** [comment]
Thanks for the review @jacoguzo - will address as soon as possible!

**Comment by melissawm** [comment]
This should now be up to date and ready for another review. Thanks!

**Review by jacoguzo — APPROVED** [review]
